### PR TITLE
Setup chromatic and cleanup storybook

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
I want to merge this change because this PR:
1) Adds [Chromatic](https://www.chromatic.com/) to every PR. Chromatic runs on our storybook - thanks to that we will have visual feedback when we introduce UI changes in PR
2) Removed not needed stories - they displayed only dialogs
3) Brings back storybook - previously it was not working
4) Updates storybook to version 7
5) Adds `any` type to the result of parsing URL query params. Why? By migrating to the storybook in version 7 we also have new types for qs package which are now not typed as `any` anymore. In order now to add more code to this PR I decided to create an issue for migrating to new types from qs package and add manual `any` instead.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -100,3 +100,35 @@ jobs:
...
+            ${{ runner.os }}-qa-
+            ${{ runner.os }}-
+
+      - name: Install deps
+        run: npm ci
+
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (NyanKiyoshi) [inline .github/workflows/test.yml:133]: Restrict token permissions. Why does it need that GH token?

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -100,3 +100,35 @@ jobs:
         run: |
           npm run extract-messages
           git diff --exit-code ./locale
+
+  chromatic-storybook:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (NyanKiyoshi) [inline .github/workflows/test.yml:106]: Consider splitting the CI into multiple files, that job doesn't appear to depend on the rest of the jobs

File: .github/workflows/test.yml (hunk 1 above)
Comment (krzysztofzuraw) [reply .github/workflows/test.yml]: You are right - I've removed `token` from `Publish to Chromatic` action

File: .github/workflows/test.yml (hunk 2 above)
Comment (krzysztofzuraw) [reply .github/workflows/test.yml]: Sure - fixed 😀

File: src/auth/index.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import ResetPassword from "./views/ResetPassword";
 import ResetPasswordSuccess from "./views/ResetPasswordSuccess";
 
 const LoginView: React.FC<RouteComponentProps<any>> = () => {
-  const qs = parseQs(location.search.substr(1));
+  const qs = parseQs(location.search.substr(1)) as any;
```
Comment (poulch) [inline src/auth/index.tsx:19]: question: why any?

File: src/products/components/ProductCreatePage/ProductCreatePage.stories.tsx [hunk 4]
```diff
@@ -19,132 +17,134 @@ import ProductCreatePage from "./ProductCreatePage";
...
+    taxClasses={taxClasses}
+    fetchMoreTaxClasses={undefined}
+    weightUnit="kg"
+    referencePages={[]}
+    referenceProducts={[]}
+    attributeValues={[]}
+    onAssignReferencesClick={() => undefined}
+    onCloseDialog={() => undefined}
+    onSelectProductType={() => undefined}
+    onAttributeSelectBlur={() => undefined}
```
Comment (poulch) [inline src/products/components/ProductCreatePage/ProductCreatePage.stories.tsx:58]: question: why not store these props in a variable and pass it to component?

File: src/products/components/ProductVariantCreatePage/ProductVariantCreatePage.stories.tsx [hunk 5]
```diff
@@ -1,154 +1,158 @@
...
+    saveButtonBarState="default"
+    warehouses={warehouseList}
+    onWarehouseConfigure={() => undefined}
+    referencePages={[]}
+    referenceProducts={[]}
+    attributeValues={[]}
+    fetchAttributeValues={() => undefined}
+    onAssignReferencesClick={() => undefined}
+    onCloseDialog={() => undefined}
+    onAttributeSelectBlur={() => undefined}
```
Comment (poulch) [inline src/products/components/ProductVariantCreatePage/ProductVariantCreatePage.stories.tsx:35]: question: what about move these props to a variable?

**Review by poulch — APPROVED** [review]

File: .github/workflows/storybook-chromatic.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+name: QA
```
Comment (NyanKiyoshi) [inline .github/workflows/storybook-chromatic.yml:1]: Is the name correct?

File: .github/workflows/storybook-chromatic.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+          restore-keys: |
+            ${{ runner.os }}-qa-${{ env.cache-name }}-
+            ${{ runner.os }}-qa-
+            ${{ runner.os }}-
+
+      - name: Install deps
+        run: npm ci
+
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
```
Comment (NyanKiyoshi) [inline .github/workflows/storybook-chromatic.yml:33]: Freeze the 3rd party:

```suggestion
        uses: chromaui/action@a89b674adf766dbde41ad9ea2b2b60b91188a0f0 # v6.17.4
```

Comment (krzysztofzuraw) [reply .github/workflows/storybook-chromatic.yml:33]: I'm wondering if the same shouldn't be applied to dashboard npm dependencies. As we are using `^` and by `npm install` we catch all patch versions e.g
CleanShot 2023-05-09 at 10 52 42@2x

**Review by poulch — APPROVED** [review]

**Review by NyanKiyoshi — APPROVED** [review]
CIs: LGTM

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**